pypeit/images/combineimage.py

```diff
@@ -4,133 +4,185 @@
-                img_stack = np.zeros(shape)
-                ivar_stack= np.zeros(shape)
-                rn2img_stack = np.zeros(shape)
+                # Allocate arrays to collect data for each frame
+                shape = (self.nfiles,) + pypeitImage.shape
+                img_stack = np.zeros(shape, dtype=float)
+                ivar_stack= np.ones(shape, dtype=float)
+                rn2img_stack = np.zeros(shape, dtype=float)
                 crmask_stack = np.zeros(shape, dtype=bool)
-                # Mask
-                bitmask = imagebitmask.ImageBitMask()
                 mask_stack = np.zeros(shape, bitmask.minimum_dtype(asuint=True))
-            # Grab the lamp status
-            lampstat += [self.spectrograph.get_lamps_status(pypeitImage.rawheadlist)]
+                lampstat = [None]*self.nfiles
+
+            # Save the lamp status
+            lampstat[kk] = self.spectrograph.get_lamps_status(pypeitImage.rawheadlist)
             # Process
-            img_stack[kk,:,:] = pypeitImage.image
+            img_stack[kk] = pypeitImage.image
             # Construct raw variance image and turn into inverse variance
             if pypeitImage.ivar is not None:
-                ivar_stack[kk, :, :] = pypeitImage.ivar
-            else:
-                ivar_stack[kk, :, :] = 1.
+                ivar_stack[kk] = pypeitImage.ivar
             # Mask cosmic rays
             if pypeitImage.crmask is not None:
-                crmask_stack[kk, :, :] = pypeitImage.crmask
+                crmask_stack[kk] = pypeitImage.crmask
             # Read noise squared image
             if pypeitImage.rn2img is not None:
-                rn2img_stack[kk, :, :] = pypeitImage.rn2img
+                rn2img_stack[kk] = pypeitImage.rn2img
             # Final mask for this image
-            # TODO This seems kludgy to me. Why not just pass ignore_saturation to process_one and ignore the saturation
-            # when the mask is actually built, rather than untoggling the bit here
+            # TODO: This seems kludgy to me. Why not just pass ignore_saturation
+            # to process_one and ignore the saturation when the mask is actually
+            # built, rather than untoggling the bit here?
             if ignore_saturation:  # Important for calibrations as we don't want replacement by 0
-                indx = pypeitImage.bitmask.flagged(pypeitImage.fullmask, flag=['SATURATION'])
-                pypeitImage.fullmask[indx] = pypeitImage.bitmask.turn_off(
-                    pypeitImage.fullmask[indx], 'SATURATION')
-            mask_stack[kk, :, :] = pypeitImage.fullmask
+                indx = pypeitImage.bitmask.flagged(pypeitImage.fullmask, flag='SATURATION')
+                pypeitImage.fullmask[indx] \
+                        = pypeitImage.bitmask.turn_off(pypeitImage.fullmask[indx], 'SATURATION')
+            mask_stack[kk] = pypeitImage.fullmask
```

Comment:
Can you please go to pypeitimage.py and more clearly document in the Datamodel that fullmask is a bit mask. Also, the allowed bits are as far as I can see not documented anywhere, making these bitmasks very difficult to interpret in practice.

Reply:
There's some discussion of this here:

https://pypeit.readthedocs.io/en/release/out_masks.html

I've added a link to this doc from the "basic image processing" doc.  Can you have a look at those docs and request improvements/changes?